pkg/common/common.go

```diff
@@ -17,6 +17,7 @@ package common
 const (
 	CalicoNamespace               = "calico-system"
 	TyphaDeploymentName           = "calico-typha"
+	APIServerDeploymentName       = "calico-apiserver"
```

Comment:
Let's not add this here.

Instead, I think we should take this logic: https://github.com/tigera/operator/blob/4eff4995da85134f91a4a7c3bebf072b1095f35e/pkg/render/apiserver.go#L954-L960

And make a new function in `pkg/render/apiserver.go` like so:

```go
func deploymentName() string { . . .}
```

Reply:
Can we generalize the deployment naming like this?

```go
// Determine names based on the configured variant
// It takes two name as parameters, enterpriseName and ossName, and returns name and nameToDelete.
func (c *apiServerComponent) deploymentName(enterpriseName, ossName string) (string, string) {
    var name, nameToDelete string
    switch c.cfg.Installation.Variant {
    case operatorv1.TigeraSecureEnterprise:
        name = enterpriseName
        nameToDelete = ossName
    case operatorv1.Calico:
        name = ossName
        nameToDelete = enterpriseName
    }
    return name, nameToDelete
}

// Usage
// name, nameToDelete := c.deploymentName("tigera-apiserver", "calico-apiserver")
// name, nameToDelete := c.deploymentName("tigera-apiserver-delegate-auth", "calico-apiserver-delegate-auth")
```
This should reduce around 60 lines in `pkg/render/apiserver.go`.

The final output should look like this commit https://github.com/MuhtasimTanmoy/operator/commit/f31415c21914b695abd158641c51d2665353e37c made on a temporary branch for review. Looking for better intuitive naming btw.

Reply:
Seems like a nice cleanup to me!

The only thing I would say is that the function shouldn't be called `deploymentName` if we're going to make it generalized, since not all of those objects are Deployments. 

Maybe `r.getResourceName`?

Reply:
Yes, the function definitely should not be `deploymentName`. 

As the sole responsibility of the function is resolving name based on the configured variant regardless of it's usage, how about `c.resolveNameByVariant` or `c.resourceNameBasedOnVariant`?

Reply:
Sounds perfect :+1: